managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java

```diff
@@ -542,8 +543,17 @@ public void closeFailed(ManagedLedgerException exception, Object ctx) {
         }
 
         scheduledExecutor.shutdownNow();
-        orderedExecutor.shutdownNow();
+        if (!scheduledExecutor.awaitTermination(30, TimeUnit.SECONDS)) {
+           log.error("Cannot stop scheduledExecutor gracefully.");
+        }
+        orderedExecutor.shutdown();
```

Comment:
Why not: 
```suggestion
        orderedExecutor.shutdownNow();
```

Reply:
I have removed this change from this PR.